### PR TITLE
[SPARK-27946][SQL] Revise error message for SHOW CREATE TABLE 

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

Revises current error message so users can know what to do.

### Why are the changes needed?

Current error message isn't clear so users might be confused.

### Does this PR introduce any user-facing change?

Yes. Users should see more clear error message.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala [hunk 1]
```diff
@@ -1110,7 +1110,9 @@ case class ShowCreateTableCommand(table: TableIdentifier)
         }
 
         if (tableMetadata.tableType == VIEW) {
-          throw new AnalysisException("Hive view isn't supported by SHOW CREATE TABLE")
+          throw new AnalysisException("Hive view isn't supported by SHOW CREATE TABLE. " +
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: BTW, are all views Hive view? It looks a bit weird to treat view as Hive serde table.

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: We have a `DDLUtils.isDatasourceTable` check above. Will other view come to here?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Ideally, view is not table so SHOW CREATE TABLE should not work for views. If we don't want to add a SHOW CREATE VIEW and just want to support view in SHOW CREATE TABLE, then I think it's better to not require the AS SERDE.

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: What you mean "not require the AS SERDE"? You mean "SHOW CREATE TABLE" should work on Hive view?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: IMO there is no such thing called "Hive view". Spark has permanent views which can be stored in Hive metatore, which is no different from views created by Hive.

Furthermore, what's the Hive specify syntax for creating views? Maybe we should support views in both SHOW CREATE TABLE and SHOW CREATE TABLE AS SERDE

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: OK. I get your point now.

This is Hive syntax for creating views:

```
CREATE VIEW [IF NOT EXISTS] [db_name.]view_name [(column_name [COMMENT column_comment], ...) ]
  [COMMENT view_comment]
  [TBLPROPERTIES (property_name = property_value, ...)]
  AS SELECT ...;
```
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateView


This is Spark syntax for creating views:
```
CREATE [OR REPLACE] [[GLOBAL] TEMPORARY] VIEW [IF NOT EXISTS [db_name.]view_name
    create_view_clauses
    AS query;
```
https://spark.apache.org/docs/3.0.0-preview/sql-ref-syntax-ddl-create-view.html

Looks like it is basically the same. I think we could support views in both commands as you said.

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: If so, then I think I could close this follow-up and prepare a full PR to add the support. WDYT?

Comment (viirya) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: Btw, for a view, what `DDLUtils.isDatasourceTable`  will tell about it?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: SGTM. To check a view, the provider should be empty and the table type should be VIEW.

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala]: `DDLUtils.isDatasourceTable` and `DDLUtils.isHiveTable` both return false for views.

**Pull request closed without merging**